src/backend/optimizer/plan/planner.c

```diff
@@ -309,6 +339,8 @@ planner(Query *parse, int cursorOptions,
       {
         resource_negotiator(parse, cursorOptions, boundParams, resourceLife, &ppResult);
 
+		decrease_planning_depth();
```

Comment:
better to move decrease_planning_depth to the end of try. (just before end_memory_accounting)